### ISSUE TITLE
M4C Pretrained Model Reproduced

### ISSUE THREAD

**Issue by HenryJunW** [facebookresearch/mmf#793]
## ❓ Questions and Help
Fantastic job! 

I try to train the M4C on Textvqa & Stvqa from scratch to reproduce the pre-trained model results. However, I can only get ~39.52% on the Textvqa validation set, **1 percent below** 40.65% achieved by the given pre-trained model. I am wondering whether there is any tricks to train the pre-trained model, i.e. more epochs? It would be very helpful if you can provide some specs including GPUs #, num_workers # and so on, while getting the pre-trained model. 

The reason is that I would like to train m4c on my own custom dataset similar with Textvqa from scratch, however, I would like to know what factors to consider for getting a decent results. Thanks in advance!

**Comment by ytsheng** [comment]
Hi @HenryJunW, thanks for using mmf!

Two things:-
1. Depending on what you set as the training_seed, results may vary slightly. We recommend that you train your training run with random seed and average the results after 5 runs.
2. Are you using distributed training in your pretraining? In practice, with distributed training we also have observed some fluctuations in the results. 

Hope this helps!

**Comment by HenryJunW** [comment]
Thanks for your reply, make sense! 

I am currently using 4 V100 GPUs to do pretraining. In terms of distributed training, are you referring to that it is better to train using a single GPU? If so, do you recommend changing the lr in this case or simply using the provided config file?

**Comment by ytsheng** [comment]
I am merely commenting on that the performance numbers will differ slightly, not that it is better. :)

**Comment by apsdehal** [comment]
@HenryJunW It would be helpful if you can provide us the exact command you ran so we can better help you.

**Comment by HenryJunW** [comment]
@ytsheng  I see, thanks for the clarification!
@apsdehal  Currently, I tried to reproduced the pretrained model validation result of joint_with_stvqa model. The commands i am using is, 
```
mmf_run dataset=textvqa \
model=m4c \
config=./projects/m4c/configs/textvqa/joint_with_stvqa.yaml \
env.save_dir=./save/m4c_joint_with_stvqa_from_scratch \
env.cache_dir=./cache/m4c_joint_with_stvqa_from_scratch \
env.data_dir=./data/m4c \
run_type=train_val \
```
I didn't change the default/joint_with_stvqa yaml files except I modified num_workers to be 2 from 4 in order to fit my CPU RAM limit. The validation result I got is mentioned in the original question. Thanks for your help, really appreciate it very much. 

In the meantime, I will give a shot trying to changing seed number and get results average as recommended by @ytsheng
